### ISSUE TITLE
Cannot install Karma because of gyp errors.

### ISSUE THREAD

**Issue by jamesDrinkard** [nodejs/node-gyp#800]
I was having problems trying to install a new workspace of some code with npm with Windows 7 and I got this error:
```
gyp ERR! build error
gyp ERR! stack Error: Can't find "msbuild.exe". Do you have Microsoft Visual Stu
dio C++ 2008+ installed?
```
I have the Visual Studio Professional 2013 version installed, npm 2.10.1, bower 1.5.2, and I have the 64 bit version for nodejs.
I tried multiple things to get past this, but found this command after a search which worked:
npm install -g node-gyp
and I was able to get grunt serve to work and load my code.

However, I went to install Karma so I could do testing and I got the same type of error again:
```
gyp ERR! build error
gyp ERR! stack Error: Can't find "msbuild.exe". Do you have Microsoft Visual Stu
dio C++ 2008+ installed?
gyp ERR! stack     at findMsbuild (C:\Program Files\nodejs\node_modules\npm\node
_modules\node-gyp\lib\build.js:121:23)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node
-gyp\lib\build.js:103:11
gyp ERR! stack     at F (C:\Program Files\nodejs\node_modules\npm\node_modules\w
hich\which.js:72:25)
gyp ERR! stack     at E (C:\Program Files\nodejs\node_modules\npm\node_modules\w
hich\which.js:75:29)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\whic
h\which.js:83:16
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:95:15)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modu
les\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\<myName>\AppData\Roaming\npm\node_modules\karma\node_modu
les\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate

gyp ERR! node -v v0.12.4
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok
```
but it referenced Karma now.

After finding your site, I started reviewing the prerequisites, so I did uninstall Python 2.7.1 and then installed version 2.7.3.  Then I setup the environment variable for the path, which worked for finding the python version from the command line.  Then I installed Microsoft Visual Studio C++ 2013 for Windows Desktop Express version and then the Windows 7 64-bit SDK, I followed this instruction : WindowsSDKDir not being set, and you are sure you have already installed the SDK:

Which loaded up lots of things on my machine, but to no avail.  Is there an order with this installation process or is there something else I'm missing?  I really need to get this installed properly.

**Comment by bnoordhuis** [comment]
That error probably means you need to run vcvarsall.bat that ships with VS, it sets up the environment.  You can also start a terminal from VS that runs it for you.  There's a menu option for it somewhere.

**Comment by jamesDrinkard** [comment]
I can duplicate this error now, each time I start with a new project.  I found the file under: C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC>vcvarsall.bat, even though I'm running Visual Studio 2013.  So I tried running the bat file, but I got ERROR: Cannot determine the location of the VS Common Tools folder.  

I also found out that security put a new policy in place that prevents npm from getting to the Users folder, unless I manually add a path statement to my command prompt that lets npm find grunt and bower.  Could this be related to this problem or is it unrelated since both are path related?
